confluent/docker_utils/cub.py

```diff
@@ -261,6 +263,68 @@ def check_connect_ready(host, port, service_timeout):
         return False
 
 
+def check_ksql_server_ready(host, port, service_timeout):
+    """Waits for KSQL server to be ready.
+
+    Args:
+        host: Hostname where KSQL server is hosted.
+        port: KSQL server port.
+        timeout: Time in secs to wait for the service to be available.
+
+    Returns:
+        False, if the timeout expires and KSQL server is not ready, True otherwise.
+
+    """
+
+    # Check if you can connect to the endpoint
+    status = wait_for_service(host, port, service_timeout)
+
+    if status:
+        # Check if service is responding as expected to basic request
+        url = "http://%s:%s" % (host, port)
+        r = requests.get(url)
+        # The call should always return a json string including version
+        if r.status_code // 100 == 2 and 'version' in str(r.text):
+            return True
+        else:
+            print("Unexpected response with code: %s and content: %s" % (str(r.status_code), str(r.text)), file=sys.stderr)
+            return False
+    else:
+        print("%s cannot be reached on port %s." % (str(host), str(port)), file=sys.stderr)
+        return False
+
+
+def check_control_center_ready(host, port, service_timeout):
+    """Waits for Confluent Control Center to be ready.
+
+    Args:
+        host: Hostname where Control Center is hosted.
+        port: Control Center port.
+        timeout: Time in secs to wait for the service to be available.
+
+    Returns:
+        False, if the timeout expires and Connect is not ready, True otherwise.
+
+    """
+
+    # Check if you can connect to the endpoint
+    status = wait_for_service(host, port, service_timeout)
+
+    if status:
+        # Check if service is responding as expected to basic request
+        url = "http://%s:%s" % (host, port)
```

Comment:
How do we define "ready"? An alternative to simply checking the HTTP server is responding would be to call the C3 API: 

```
$ http http://localhost:9021/2.0/health/status|jq '.clusterStatus'
{
  "A2xvXUXdTUKBhaQjdgYZPg": {
    "metric": "CLUSTER_STATUS",
    "value": "ONLINE",
    "timestamp": 1555405331246
  }
}
```

and poll for the `CLUSTER_STATUS` metric where `value` is `ONLINE`

Reply:
A benefit of this could be that users don't then launch C3 whilst the streams apps are still loading and get the C3 loading page (which IIRC they will if the HTTP server is up but other stuff is still initialising)

Reply:
Two reasons to not do that:

1. The definition of `ready` for this purpose is when the API can respond to API calls to do things like https://github.com/confluentinc/cp-demo/blob/5.2.1-post/scripts/start.sh#L66 (which renames the cluster).
2. Back to concern mentioned above wrt `KsqlServerInfo`, the path may change or the response format and strings may change.  In anticipation of what we don't know, the benefit is keeping it more generalized to "future proof" against potential changes